tests/components/tankerkoenig/test_coordinator.py

```diff
@@ -50,3 +56,70 @@ async def test_rate_limit(
     state = hass.states.get("binary_sensor.station_somewhere_street_1_status")
     assert state
     assert state.state == "on"
+
+
+@pytest.mark.parametrize(
+    ("exception", "expected_log"),
+    [
+        (
+            TankerkoenigInvalidKeyError,
+            "invalid key error occur during update of stations",
+        ),
+        (
+            TankerkoenigRateLimitError,
+            "API rate limit reached, consider to increase polling interval",
+        ),
+        (TankerkoenigConnectionError, "error occur during update of stations"),
+        (TankerkoenigError, "error occur during update of stations"),
+    ],
+)
+@pytest.mark.usefixtures("setup_integration")
+async def test_update_exception_logging(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    tankerkoenig: AsyncMock,
+    caplog: pytest.LogCaptureFixture,
+    exception: None,
+    expected_log: str,
+) -> None:
+    """Test log messages about exceptions during update."""
+    tankerkoenig.prices.side_effect = exception
+    async_fire_time_changed(
+        hass, dt_util.utcnow() + timedelta(minutes=DEFAULT_SCAN_INTERVAL)
+    )
+    await hass.async_block_till_done()
+    assert expected_log in caplog.text
+    state = hass.states.get("binary_sensor.station_somewhere_street_1_status")
+    assert state
+    assert state.state == STATE_UNAVAILABLE
+
+
+@pytest.mark.parametrize(
+    ("exception", "expected_log"),
+    [
+        (
+            TankerkoenigInvalidKeyError,
+            "invalid key error occur during setup of station",
+        ),
+        (TankerkoenigConnectionError, "connection error occur during setup of station"),
+        (TankerkoenigError, "Error when adding station"),
+    ],
+)
+@pytest.mark.usefixtures("tankerkoenig")
+@pytest.mark.usefixtures("config_entry")
```

Comment:
You can remove this, as you passe them already as params

Reply:
you're right ... removed them 🙂